jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/AbstractDiagnosticsCollector.java

```diff
@@ -0,0 +1,301 @@
+	    		return true;
+	    	// only check fully qualified annotations 
+	    	if (annotationFQName.equals(elementName)) {
+	    		IJavaElement parent = annotation.getParent();
+		        IType declaringType = (parent instanceof IType) ? (IType) parent : 
+		        	((parent instanceof IMember) ? ((IMember) parent).getDeclaringType() : null) ;
+		    	if (declaringType != null) { 
+					String[][] fqName = declaringType.resolveType(elementName);	// the call could be expensive
+					if (fqName != null && fqName.length == 1) {
+						return annotationFQName.equals(JavaModelUtil.concatenateName(fqName[0][0], fqName[0][1]));
+					}
+		    	}
+	    	}
+    	}
+    	return false;
+    }
+    
+	/**
+	 * Returns true if the java element name matches the given fully qualified java element name and
+	 * false otherwise.
+	 *
+	 * @param unit    	 		compilation unit of Java class.
+	 * @param annotation    	given annotation object.
+	 * @param annotationFQName	the fully qualified annotation name.
+	 * @return true if the java element name matches the given fully qualified java element name and false otherwise.
+	 */
+    protected static boolean isMatchedJavaElement(IType type, String javaElementName, String javaElementFQName) throws JavaModelException {    	
+    	if (javaElementFQName.endsWith(javaElementName)) {
+			// For performance reason, we check if the import of annotation name is
+			// declared
+	    	if (isImportedJavaElement(type.getCompilationUnit(), javaElementFQName) == true)
+	    		return true;
+	    	// only check fully qualified java element 
+	    	if (javaElementFQName.equals(javaElementName)) {
+				String[][] fqName = type.resolveType(javaElementName);	// the call could be expensive
+				if (fqName != null && fqName.length == 1) {
+					return javaElementFQName.equals(JavaModelUtil.concatenateName(fqName[0][0], fqName[0][1]));
+				}
+	    	}
+    	}
+    	return false;
+    }    
+        
+	/**
+	 * Returns true if the given Java class imports the given Java element and
+	 * false otherwise.
+	 *
+	 * @param type    	 			Java class.
+	 * @param javaElementFQName		given Java element fully qualified name.
+	 * @return true if the Java class imports the given Java element and false otherwise.
+	 */
+    protected static boolean isImportedJavaElement(ICompilationUnit unit, String javaElementFQName) throws JavaModelException {
+		IImportContainer container = unit.getImportContainer();
+		if (container == null) {
+			return false;
+		}
+		
+		// The following code uses JDT internal class and looks like
+		// ICompilationUnit#getImports()
+		// To avoid creating an array of IImportDeclaration, we do the following code:
+		JavaModelManager manager = JavaModelManager.getJavaModelManager();
+		Object info = manager.getInfo(container);
+		if (info == null) {
+			if (manager.getInfo(unit) != null) {
+				// CU was opened, but no import container, then no imports
+				// return NO_IMPORTS;

```

Comment:
possibly an old return statement that wasn't cleaned up?

Reply:
Do we need this return statement? Or what does the comment mean?

Reply:
No, already been removed

Reply:
I still see the commented out `// return NO_IMPORTS;` lines in this class

Reply:
it's missed in the last commit